mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageDAO.java

```diff
@@ -381,4 +400,65 @@ public boolean isFound() {
             return message.get();
         }
     }
+
+    public CompletableFuture<Stream<MessageIdAttachmentIds>> retrieveAllMessageIdAttachmentIds() {
+        return cassandraAsyncExecutor.execute(
+            selectAllMessagesWithAttachment.bind()
+                .setReadTimeoutMillis(configuration.getMessageAttachmentIdsReadTimeout()))
+            .thenApply(resultSet -> cassandraUtils.convertToStream(resultSet)
+                .map(this::fromRow)
+                .filter(MessageIdAttachmentIds::hasAttachment));
+    }
+
+    private MessageIdAttachmentIds fromRow(Row row) {
+        MessageId messageId = messageIdFactory.of(row.getUUID(MESSAGE_ID));
+        Set<AttachmentId> attachmentIds = attachmentByIds(row.getList(ATTACHMENTS, UDTValue.class))
+            .map(MessageAttachmentRepresentation::getAttachmentId)
+            .collect(Guavate.toImmutableSet());
+        return new MessageIdAttachmentIds(messageId, attachmentIds);
+    }
+
+    public static class MessageIdAttachmentIds {
+        private final MessageId messageId;
+        private final Set<AttachmentId> attachmentIds;
+        
+        public MessageIdAttachmentIds(MessageId messageId, Set<AttachmentId> attachmentIds) {
```

Comment:
if it's public, there's some things to enforce : not null args, immutable things, etc

Reply:
ok